### PR TITLE
Expose parse_os, parse_device and parse_ua

### PR DESCRIPTION
**Pull request by daniel-statsig**
I have a use case where I only care about individual parts of the UserAgent. I am looking for performance gains and noticed that multiple regexes are done and the results are not being used.

If it is possible, would you mind exposes these functions as public?

My code that uses the parser is here (https://github.com/statsig-io/ruby-sdk/blob/main/lib/evaluator.rb#L453-L461). I hope this better illustrates what I am attempting to do.

### CODE REVIEW

**Comment by daniel-statsig** [comment]
These are actually public in the uap-java library too https://github.com/ua-parser/uap-java/blob/master/src/main/java/ua_parser/Parser.java#L89-L99

File: lib/user_agent_parser/parser.rb [hunk 1]
```diff
@@ -39,6 +39,36 @@ def parse(user_agent)
...
+      pattern, match = first_pattern_match(@device_patterns, user_agent)
+
+      if match
+        device_from_pattern_match(pattern, match)
+      else
+        Device.new
+      end
+    end
+
+    def parse_ua(user_agent, os = nil, device = nil)
```
Comment (opti) [inline lib/user_agent_parser/parser.rb:62]: @daniel-statsig I think the duplicated private method can be deleted. https://github.com/ua-parser/uap-ruby/blob/241085538c3ba165c476d63f48c5886213412088/lib/user_agent_parser/parser.rb#L109-L117

**Review by opti — CHANGES_REQUESTED** [review]
Thank you for the contribution!
Would you mind adding some specs as well to those methods you've made public?

File: lib/user_agent_parser/parser.rb (hunk 1 above)
Comment (daniel-statsig) [reply lib/user_agent_parser/parser.rb:62]: Oh sorry, I missed that. Will remove

**Comment by daniel-statsig** [comment]
@opti, I have addressed your feedback. Thank you

**Review by opti — APPROVED** [review]

**Comment by opti** [comment]
Thank you @daniel-statsig, new version `v2.15.0` is now released.

**Pull request merged**